### ISSUE TITLE
Roundtrip fuzz test fails

### ISSUE THREAD

**Issue by Janiczek** [Airsequel/double-x-encoding#1]
In the Elm implementation, when doing a roundtrip test
```elm
    Test.fuzz Fuzz.string "Roundtrip" <|
        \str ->
            str
                |> DoubleXEncode.doubleXEncode
                |> DoubleXEncode.doubleXDecode
                |> Expect.equal str
```
the test fails:

```
    "𐀀a"
    ╷
    │ Expect.equal
    ╵
    "􀀀"
```

In case the Unicode gets lost on the way, these are the `Char.toCode` integers:
```
    [65536,97]
    ╷
    │ Expect.equal
    ╵
    [1048576]
```

**Comment by ad-si** [comment]
Hi, thanks for the heads up. I know that the Elm version isn't quite right yet, but so far I couldn't make any sense of it 😔. When you check it with the [BLNS](https://github.com/minimaxir/big-list-of-naughty-strings) there are also 3 of 515 erroneous.
(Replace the Blns.elm file with following file [Blns.elm](https://github.com/Airsequel/double-x-encoding/files/10406683/Blns.elm.txt), run `cd Elm && elm reactor`, and then open `http://localhost:8000/Test.elm`)

<img width="502" alt="image" src="https://user-images.githubusercontent.com/36796532/212196873-0d842cc2-0b6e-4479-8b36-43c494138868.png">

Any help would be highly appreciated! 🙇‍♂️

**Comment by ad-si** [comment]
Elm seems to use `fromCharCode()` and not `codePointAt()` (like it should ^^): https://github.com/elm/core/blob/65cea00afa0de03d7dda0487d964a305fc3d58e3/src/Elm/Kernel/Char.js#L18
Maybe this implementation is faulty 😳

**Comment by ad-si** [comment]
Haha, you opened exactly that issue 6 years ago: https://github.com/elm/core/pull/687 😂

**Comment by ad-si** [comment]
> Maybe this implementation is faulty 😳

It doesn't seem to be faulty when I run each character of the BLNS through it:
https://replit.com/@ad_si/Test-Elms-toCode-and-fromCode#index.js

**Comment by Janiczek** [comment]
The bug was closed as fixed back then. I didn't dive too deep into this issue so I don't understand where the problem is. But it's possible some Elm stdlib functions need some more tweaking still.

**Comment by ad-si** [comment]
Found the problem: It doesn’t support characters from the Supplementary Private Use Area-B Unicode Plane, since characters there can be 6 hex digits long. It’s annoying that I have to support it since characters there literally can’t be typed in any meaningful way, but I’ll add support for a `XXYaaaaaa` pattern to specify those characters in a backwards compatible way and without making the encoding longer for common usage.

**Comment by Janiczek** [comment]
> It doesn’t support characters from the Supplementary Private Use Area-A Unicode Plane

Sorry, just to make sure - `It` meaning `double-x-encoding`? or rather Elm's `String`/`Char` modules?
Just so that I know how to update my mental model 😁

**Comment by ad-si** [comment]
Sorry 😅, "it" meaning Double-X-Encoding. It's an omission in my code 🙈, the Elm implementation of String / Char seems to be fine as far as my testing went.

**Comment by ad-si** [comment]
Fixed it in a911cda. Thanks again for posting the failing test @Janiczek. It was the clue I needed to get to the bottom of this! 😊
